devito/ir/support/space.py

```diff
@@ -148,9 +155,49 @@ def _rebuild(self):
 
     @cached_property
     def size(self):
-        upper_extreme = self.dim.symbolic_max + self.upper
-        lower_extreme = self.dim.symbolic_min + self.lower
-        return (upper_extreme - lower_extreme + 1) / self.dim.symbolic_incr
+        # The Interval size is typically a function of several symbols (e.g.,
```

Comment:
why a comment instead of docstring?

Reply:
fair point. Class docstrings now refactored.